### PR TITLE
Added API calls for workspaces and projects lookup

### PR DESCRIPTION
**Pull request by Raalsky**

### CODE REVIEW

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 1]
```diff
@@ -190,6 +191,47 @@ def get_project(self, project_id: str) -> Project:
...
+    def get_available_projects(self,
+                               workspace_id: Optional[str] = None,
+                               search_term: Optional[str] = None
+                               ) -> List[Project]:
+        try:
+            response = self.backend_client.api.listProjects(
+                limit=5,
+                organizationIdentifier=workspace_id,
+                searchTerm=search_term,
+                sortBy=['timeOfCreation'],
```
Comment (shnela) [inline neptune/new/internal/backends/hosted_neptune_backend.py:204]: Shouldn't we rely on `last_experiment_creation_time` instead?

File: neptune/new/internal/backends/hosted_neptune_backend.py [hunk 2]
```diff
@@ -190,6 +191,47 @@ def get_project(self, project_id: str) -> Project:
...
+                sortDirection=['descending'],
+                userRelation='memberOrHigher',
+                **self.DEFAULT_REQUEST_KWARGS,
+            ).response()
+            warning = response.metadata.headers.get('X-Server-Warning')
+            if warning:
+                click.echo(warning)  # TODO print in color once colored exceptions are added
```
Comment (shnela) [inline neptune/new/internal/backends/hosted_neptune_backend.py:211]: What colored exceptions depend on?

File: neptune/new/internal/backends/hosted_neptune_backend.py (hunk 1 above)
Comment (Raalsky) [reply neptune/new/internal/backends/hosted_neptune_backend.py:204]: I've talked with Marcin and the expected behavior is to sort by both last experiment creation and project creation :wink:

File: neptune/new/internal/backends/hosted_neptune_backend.py (hunk 2 above)
Comment (Raalsky) [reply neptune/new/internal/backends/hosted_neptune_backend.py:211]: Dunno. Copy-pasting from old `get_project`

File: neptune/new/internal/backends/hosted_neptune_backend.py (hunk 1 above)
Comment (Raalsky) [reply neptune/new/internal/backends/hosted_neptune_backend.py]: I've changed it to `lastViewed` due to lack of other options in `sortBy` argument without API updates. Approved by Marcin.

**Review by shnela — APPROVED** [review]

**Pull request merged**